224-basic-calculator.py

```diff
@@ -0,0 +1,61 @@
+# 224. Basic Calculator
+
+class Solution:
+    def calculate(self,s):
+        """
+        :type s: str
+        :rtype: int
+        """
+        tokens = tokenizeWithParens(s)
+        # print(tokens)
+        stack = []
+        for t in tokens:
+            if t.isdigit():
+                stack.append(int(t))
+            elif t in '(+-':
+                stack.append(t)
+            elif t == ')':
+                # print(stack)
+                subStack = []
+                while stack[-1] != '(':
+                    num = stack.pop()
+                    op = stack.pop()
+                    if op == '+':
+                        subStack.append(num)
+                    elif op == '-':
+                        subStack.append(-1 * num)
+                    elif op == '(':
+                        subStack.append(num)
+                        break
+                # print(sum(subStack))
+                stack.append(sum(subStack))
+        subStack = []
+        while stack:
+            num = stack.pop()
+            op = '+'
+            if stack:
+                op = stack.pop()
```

Comment:
I think this would be better written as
`op = stack.pop() if stack else '+'`

It's just a little clearer on one line so the reader doesn't have to wonder for a moment why you always make op +

Reply:
Also, a computation will look like this: 'number op number op number op number', always starting and ending with number. I still think it would be cleaner to do it the way I showed you on the last problem, where you start the substack with the first number and then always get the next op and next number -- no need to check the size of the stack.

Reply:
agreed, reworked the solution so that I no longer have to check for the size of the stack

Reply:
Can I merge this? lol